src/transformers/models/swin/configuration_swin.py

```diff
@@ -0,0 +1,143 @@
+
+from ...configuration_utils import PretrainedConfig
+from ...utils import logging
+from torch import nn
+
+
+logger = logging.get_logger(__name__)
+
+SWIN_PRETRAINED_CONFIG_ARCHIVE_MAP = {
+    "microsoft/swin-base": "https://huggingface.co/microsoft/swin-base patch4-window12-384/resolve/main/config.json",
+    # See all Swin models at https://huggingface.co/models?filter=swin
+}
+
+
+class SwinConfig(PretrainedConfig):
+    r"""
+    This is the configuration class to store the configuration of a [`~SwinModel`].
+    It is used to instantiate an Swin model according to the specified arguments, defining the model
+    architecture. Instantiating a configuration with the defaults will yield a similar configuration to that of
+    the Swin [swin-base](https://huggingface.co/swin-base) architecture.
+
+    Configuration objects inherit from  [`PretrainedConfig`] and can be used
+    to control the model outputs. Read the documentation from  [`PretrainedConfig`]
+    for more information.
+
+
+    Args:
+        image_size (`int`, *optional*, defaults to 224):
+            Size (resolution) of the input images.
+        patch_size (`int`, *optional*, defaults to 4):
+            Size of patches.
+        num_channels (`int`, *optional*, defaults to 3):
+            Number of channels in input images.
+        num_labels (`int`, *optional*, defaults to 1000):
+            Number of classes for classification head
+        embed_dim (`int`, *optional*, defaults to 96):
+            Dimensionality of patch embedding.
+        depths (`tuple(int)`, *optional*, defaults to (2, 2, 6, 2)):
+            Depth of each layer in the Transformer encoder.
+        num_heads (`tuple(int)`, *optional*, defaults to (3, 6, 12, 24)):
+            Number of attention heads in each layer of the Transformer encoder.
+        window_size (`int`, *optional*, defaults to 7):
+            Size of windows.
+        mlp_ratio (`float`, *optional*, defaults to 4.0):
+            Ratio of MLP hidden dimesionality to embedding dimensionality.
+        qkv_bias (`bool`, *optional*, defaults to True):
+            Whether or not learnable bias should be added to query, key, value
+        drop_rate (`float`, *optional*, defaults to 0.0):
+            Dropout rate.
+        attn_drop_rate (`float`, *optional*, defaults to 0.0):
+            Attention dropout rate.
+        drop_path_rate (`float`, *optional*, defaults to 0.1):
+            Stochastic depth rate.
+        norm_layer (`nn.Module`, *optional*, defaults to nn.LayerNorm):
+            Normalization layer.
+        hidden_act (`str` or `function`, *optional*, defaults to `"gelu"`):
+            The non-linear activation function (function or string) in the encoder and pooler. If string, `"gelu"`,
+            `"relu"`, `"selu"` and `"gelu_new"` are supported.
+        ape (`bool`, *optional*, defaults to False):
+            Whether or not to add absolute position embedding to the patch embedding.
```

Comment:
```suggestion
        use_absolute_embeddings (`bool`, *optional*, defaults to False):
            Whether or not to add absolute position embedding to the patch embedding.
```
I would rename this to make it a bit more clear.